tests/crypto/read_file.c

```diff
@@ -228,12 +228,8 @@ oe_result_t read_mod(char* filename, uint8_t* mod, size_t* mod_size)
 
 oe_result_t read_mixed_chain(char* chain, char* chain1, char* chain2)
 {
-    char chain_temp[max_cert_chain_size * 2];
-    OE_UNUSED(chain);
-
-    strcat(chain_temp, chain1);
-    strcat(chain_temp, chain2);
-    chain = chain_temp;
+    strcat(chain, chain1);
```

Comment:
The earlier code seems like a bug. chain_temp was not being initialized to a null terminated string.
And the input pointer was just being assigned to a local variable.

What is read_mixed_chain testing?

Reply:
The earlier code was definitely returning garbage, which causes the `test_mixed_chain` in rsa_tests.c to fail when updated to check for a more specific error code. `read_mixed_chain` is just a helper function for combining two chains, and as used by `test_mixed_chain`; from the code, it is supposed to check that we reject a cert chain presented as `{ca1, root1, ca2, root2}`.